src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.CatDao;
+import core.basesyntax.dao.CatDaoImpl;
+import core.basesyntax.exception.InjectionException;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException,
+            InjectionException {
+
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+
+        try {
+            for (Field field : declaredFields) {
+                if (field.getDeclaredAnnotation(Inject.class) != null) {
+                    if (field.getType().equals(BetDao.class)
+                            && BetDaoImpl.class.getAnnotation(Dao.class) != null
```

Comment:
btw, ти ще так можеш перевіряти наявність аннотації 
```suggestion
                            && BetDaoImpl.isAnnotationPresent(Dao.class)
```